### PR TITLE
use only one implementation for beeper feedback

### PR DESCRIPTION
**Pull request by chris-bo**
I can't  test with hd44780 display, but I basicly used the old code.
To disable buzzer use 0 ms as duration
#1687 #1685

### CODE REVIEW

**Comment by AnHardt** [comment]
Compiles and beeps with REPRAP_DISCOUNT_SMART_CONTROLLER and REPRAP_DISCOUNT_FULL_GRAPHIC_SMART_CONTROLLER.

Interesting is that the beep on the ...FULL_GRAPHIC... is much louder - but this is same as before.

While you are at it.
Could be nice to have not only a feedback tone but also an additional alarm-tone when something is going wrong.

**Comment by chris-bo** [comment]
I think the "stop()" and "kill()" functions are good points to trigger an alarm tone. 
Or even everytime the "SERIAL_ERROR_START" macro is called? But as far as I can see, every serious error ends up in "stop()" or "kill()" anyway.

File: Marlin/ultralcd.cpp [hunk 1]
```diff
@@ -1137,7 +1137,28 @@ menu_edit_type(unsigned long, long5, ftostr5, 0.01)
...
+#endif
+	const unsigned int delay = 1000000 / LCD_FEEDBACK_FREQUENCY_HZ / 2;
+	int8_t i = LCD_FEEDBACK_FREQUENCY_DURATION_MS * LCD_FEEDBACK_FREQUENCY_HZ / 1000;
+	while (i--) {
+	WRITE(BEEPER,HIGH);
+	delayMicroseconds(delay);
+	WRITE(BEEPER,LOW);
+	delayMicroseconds(delay);
+}
+#endif
```
Comment (thinkyhead) [inline Marlin/ultralcd.cpp]: With coding standards applied, all conditional code should indent, including compiler directives. So this block should probably look like this instead (also re-jiggered dealing with undefined config values):

```
  #ifdef LCD_USE_I2C_BUZZER
    #ifndef LCD_FEEDBACK_FREQUENCY_HZ
      #define LCD_FEEDBACK_FREQUENCY_HZ 100
    #endif
    #ifndef LCD_FEEDBACK_FREQUENCY_DURATION_MS
      #define LCD_FEEDBACK_FREQUENCY_DURATION_MS (1000/6)
    #endif
    lcd_buzz(LCD_FEEDBACK_FREQUENCY_DURATION_MS, LCD_FEEDBACK_FREQUENCY_HZ);
  #elif defined(BEEPER) && BEEPER > -1
    #ifndef LCD_FEEDBACK_FREQUENCY_HZ
      #define LCD_FEEDBACK_FREQUENCY_HZ 500
    #endif
    #ifndef LCD_FEEDBACK_FREQUENCY_DURATION_MS
      #define LCD_FEEDBACK_FREQUENCY_DURATION_MS 50
    #endif
    SET_OUTPUT(BEEPER);
    const unsigned int delay = 1000000 / LCD_FEEDBACK_FREQUENCY_HZ / 2;
    int8_t i = LCD_FEEDBACK_FREQUENCY_DURATION_MS * LCD_FEEDBACK_FREQUENCY_HZ / 1000;
    while (i--) {
      WRITE(BEEPER,HIGH);
      delayMicroseconds(delay);
      WRITE(BEEPER,LOW);
      delayMicroseconds(delay);
    }
  #endif
```

File: Marlin/Configuration.h [hunk 2]
```diff
@@ -579,8 +579,9 @@ const bool Z_MAX_ENDSTOP_INVERTING = false; // set to true to invert the logic o
 //#define ENCODER_STEPS_PER_MENU_ITEM 5 // Set according to ENCODER_PULSES_PER_STEP or your liking
 //#define ULTIMAKERCONTROLLER //as available from the Ultimaker online store.
 //#define ULTIPANEL  //the UltiPanel as on Thingiverse
-//#define LCD_FEEDBACK_FREQUENCY_HZ 1000	// this is the tone frequency the buzzer plays when on UI feedback. ie Screen Click
-//#define LCD_FEEDBACK_FREQUENCY_DURATION_MS 100 // the duration the buzzer plays the UI feedback sound. ie Screen Click
+#define LCD_FEEDBACK_FREQUENCY_HZ 1000	// this is the tone frequency the buzzer plays when on UI feedback. ie Screen Click
+#define LCD_FEEDBACK_FREQUENCY_DURATION_MS 100 // the duration the buzzer plays the UI feedback sound. ie Screen Click
```
Comment (thinkyhead) [inline Marlin/Configuration.h]: I'm not sure if these values would be compatible with `LCD_USE_I2C_BUZZER`, so they might be best left commented out.

**Pull request merged**